### PR TITLE
Abstractify basic walls

### PR DESCRIPTION
**Pull request by ShnitzelX2**
#### Summary
Infrastructure "abstractify basic walls"

#### Purpose of change

Colored wall bloat is egregious, as is redefining the same flags over and over

#### Describe the solution

Set up a simple wall hierarchy by material

Also, defined values for fields in map_common_bash_info and related structs' constructors. `copy-from` uses the default constructor, and unset fields default to values inconsistent with the default `optional` values from load() functions, so copying from an abstract terrain didn't work.

I didn't do half-walls here because they have differences that seemed questionable

#### Testing

- Can burn t_wall but not t_concrete_wall
- Can destroy t_wall with a pickaxe but not t_concrete_wall
- Can mine t_wall and t_concrete_wall with a pickaxe
- t_wall_glass is still transparent
- Can remove paint from t_wall_blue with a paint chipper
- Deconstructed a table
- Deconstructed t_wall
- Blew up the side of a TCLF without errors
 

#### Additional context

Let me know if I misplaced anything. I'd like to do this with windows, but they need extra work for `copy-from`

### CODE REVIEW

File: data/json/furniture_and_terrain/terrain-walls.json [hunk 1]
```diff
@@ -16,27 +16,103 @@
   },
   {
     "type": "terrain",
-    "id": "t_wall",
-    "name": "wall",
-    "description": "A standard wall consisting of a wooden support structure filled with insulation and drywalled.  Painted a neutral off-white or cream color, it could use some more vibrant paint.  Looks flammable.",
-    "symbol": "#",
+    "abstract": "t_abstract_column",
+    "name": "abstract column",
+    "description": "Encompasses rock columns.  If you see this in-game, it's a bug.",
```
Comment (Night-Pryanik) [inline data/json/furniture_and_terrain/terrain-walls.json:21]:
```suggestion
    "description": { "str": "Encompasses rock columns.  If you see this in-game, it's a bug.", "//~": "NO_I18N" },
```

Comment (Night-Pryanik) [reply data/json/furniture_and_terrain/terrain-walls.json:21]: Please also apply this `NO_I18N` to all other abstracts.

Comment (ShnitzelX2) [reply data/json/furniture_and_terrain/terrain-walls.json:21]: I tried doing this with just `description`, it didn't work, and then with just `name` (like with the `ammolink` abstract) and that didn't work either. Do I need both?

Comment (ShnitzelX2) [reply data/json/furniture_and_terrain/terrain-walls.json:21]: nvm, the translation parser was the issue

**Pull request merged**